Important Updates to gmsh and gmESSI packages !!!

SumeetSinha opened the pull request:
Removed "gmsh-gmESSI Tools package" and adding two independent packages 

1) gmESSI-Tools -> Real ESSI Simulator (University of California,Davis ESSI ) plugin for auto-translation of mesh file to .fei file formats. More details about the program can be checked at http://sokocalo.engr.ucdavis.edu/~jeremic/Real_ESSI_Simulator/

2)  gmsh-Tools -> Syntax Plugin and auto-completion  for mesh generator [gmsh]. More details can be checked at http://gmsh.info/

FichteFoll commented:
You might want to provide an update with an update message about this for the original package first?
See https://packagecontrol.io/docs/messaging.

SumeetSinha commented:
Thanks for informing me. I will add it to the plugin.

FichteFoll commented:
Just notify me after the update message has been out for a while, probably around 2 weeks.

SumeetSinha commented:
Hey can you now merge my commit?. It has now been more than 2 weeks :+1:  since I chnaged.

SumeetSinha commented:
@FichteFoll any updates!!

FichteFoll commented:
Patience, please!

You have never shown any update message for "gmESSI-gmsh Tools", have you? There have been no changes to https://github.com/SumeetSinha/gmESSI-SublimePlugin in two months, especially no new tag.

You have to push a valid [semantic version](http://semver.org) [tag](http://git-scm.com/book/en/v2/Git-Basics-Tagging) (with optional leading `v`) to both of your new repos.

FichteFoll commented:
@SumeetSinha ping

SumeetSinha commented:
@FichteFoll I have added the messages but I don't see them popping while I install the plug-in. I have also added the version tag to the new plug-ins.

FichteFoll commented:
Keys in `messages.json` should not have prefixes, e.g. `v`. This *might* prevent any message from being displayed at all, but I don't know about that.

Regarding the new packages, [semantic versions](http://semver.org) consist of exactly three numeric parts.

SumeetSinha commented:
@FichteFoll I have allready  updated the versioning of the new plugins but the messages in the old plug-in  still seems not working

FichteFoll commented:
I already explained to you that you need to remove the `v`s in [messages.json](https://github.com/SumeetSinha/gmESSI-SublimePlugin/blob/2ebf4827c9d4ff14e40e04e592ce4106bf326503/messages.json).

wbond commented:
@SumeetSinha Don't forget to remove the `v` from the message.json file.

SumeetSinha commented:
Thank you @wbond and @FichteFoll. Now it is fixed :+1:

SumeetSinha commented:
Hey guys, I figured out something interesting, after struggling a lot with the messages.json file. I found out that a comma **,** at the last line was resulting in all the problem. What I mean to say is this 
1. {
	"install": "messages/install.txt",
	"2.0.0": "messages/200.txt",
	"2.0.1": "messages/201.txt"
  }
2. {
	"install": "messages/install.txt",
	"2.0.0": "messages/200.txt",
	"2.0.1": "messages/201.txt",
  }

Among 1st and 2nd version of messages.json file, 1st one pops the message whereas the 2nd one doesn't (because of the extra **,** at the end). 

I don't know whether this error was meant to be intentional or its a fault of  sublime parser.

wbond commented:
Ah, yes, the second is technically not valid JSON, since you can't have trailing commas.

SumeetSinha commented:
I think now, everything is fixed from my side. The old plugin now pops messages and the new plugins in place have valid version tags and messages. 

Thank again for helping me out.

SumeetSinha commented:
Hey can you now merge my commit?. It has now been around  2 weeks 👍 since I chnaged

SumeetSinha commented:
Anyone there !!!!

SumeetSinha commented:
@wbond . I think now this pull request is good to go ...
